tools/src/main/scala/com/fulcrumgenomics/str/bam/ReadGroupPerDuplexMolecularId.scala

```diff
@@ -0,0 +1,156 @@
+ *
+ */
+package com.fulcrumgenomics.str.bam
+
+import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.bam.Bams
+import com.fulcrumgenomics.bam.api.{SamRecord, SamSource, SamWriter}
+import com.fulcrumgenomics.cmdline.ClpGroups
+import com.fulcrumgenomics.commons.CommonsDef.IteratorToJavaCollectionsAdapter
+import com.fulcrumgenomics.commons.io.Io
+import com.fulcrumgenomics.commons.util.LazyLogging
+import com.fulcrumgenomics.sopt.{arg, clp}
+import com.fulcrumgenomics.str.FgStrDef.toMolecularId
+import com.fulcrumgenomics.str.cmdline.FgStrTool
+import com.fulcrumgenomics.umi.ConsensusTags
+import com.fulcrumgenomics.util.ProgressLogger
+import htsjdk.samtools.util.{Interval, IntervalList, OverlapDetector}
+import htsjdk.samtools.{SAMFileHeader, SAMReadGroupRecord}
+
+import scala.collection.mutable
+
+@clp(group=ClpGroups.VcfOrBcf, description=
+  """
+    |Creates a BAM file with a read group per duplex molecule ID.
+    |
+    |The raw reads should have been grouped using fgbio's GroupReadsByUmi.
+    |
+    |The duplex molecular ID for each read should be stored in the given tag (ex. MI).  A new read group per molecular
+    |ID will be written to the output BAM file.  The trailing `/A` or `/B` will be removed for duplex reads.
+    |
+    |The input file should have a single read group, with all fields from this read group copied to each new read group,
+    |besides the ID.  The read group ID for reads from a source molecule will be the original read group ID with the
+    |source molecule's ID appended (with a dash separator).  The sample name and library will have the molecular ID
+    |appended (also with a dash separator).
+    |
+    |An interval list can be given to filter output reads to a set of intervals.
+  """)
+class ReadGroupPerDuplexMolecularId
+(
+  @arg(flag='i', doc="Input SAM or BAM file (from GroupReadsByUmi).") val input: PathToBam,
+  @arg(flag='o', doc="Output SAM or BAM.") val output: PathToBam = Io.StdOut,
+  @arg(flag='r', doc="Reference sequence fasta file.") val ref: Option[PathToFasta] = None,
+  @arg(flag='l', doc="Optional list of intervals to reads to.") val intervals: Option[PathToIntervals] = None,
+  @arg(flag='T', doc="The output tag from UMI grouping.") val assignTag: String = ConsensusTags.MolecularId
+) extends FgStrTool with LazyLogging {
+
+  Io.assertReadable(input)
+  intervals.foreach(Io.assertReadable)
+  Io.assertCanWriteFile(output)
+
+  override def execute(): Unit = {
+    // Pass 1: find all the molecular identifiers to use
+    logger.info("Extracting molecular ids")
+    val mids = {
+      val progress = ProgressLogger(this.logger, unit=5e5.toInt, verb="Extracted molecule ids")
+      val in       = SamSource(input, ref=ref)
+      val iterator = toIterator(in)
+      val ids      = new mutable.HashSet[Long]()
+      iterator.foreach { rec =>
+        ids += toMolecularId(rec, this.assignTag)
+        progress.record(rec)
+        if (progress.getCount % 5e5.toInt == 0) {
```

Comment:
you can just do `if (progress.record(rec)) ....` since it returns Boolean with true = it printed something and false = it didn't.